File: app/src/main/java/com/scheffer/erik/videogamewishlist/converters/GameConverter.java
package com.scheffer.erik.videogamewishlist.converters;

import android.content.ContentValues;

import com.scheffer.erik.videogamewishlist.database.WishlistContract.GameEntry;
import com.scheffer.erik.videogamewishlist.models.Game;
import com.scheffer.erik.videogamewishlist.models.Genre;
import com.scheffer.erik.videogamewishlist.models.Platform;
import com.scheffer.erik.videogamewishlist.models.Theme;
import com.scheffer.erik.videogamewishlist.models.Video;

public class GameConverter {
    public static ContentValues toContentValues(Game game) {
        ContentValues contentValues = new ContentValues();

        contentValues.put(GameEntry.COLUMN_EXTERNAL_ID, game.getId());
        contentValues.put(GameEntry.COLUMN_NAME, game.getName());
        contentValues.put(GameEntry.COLUMN_SUMMARY, game.getSummary());
        contentValues.put(GameEntry.COLUMN_RATING, game.getRating());

        if (game.getCover() != null) {
            contentValues.put(GameEntry.COLUMN_COVER_URL, game.getCover().getUrl());
            contentValues.put(GameEntry.COLUMN_COVER_CLOUDINARY_ID, game.getCover().getCloudinaryId());
        }

        StringBuilder platformsBuilder = new StringBuilder();
        for (Platform platform : game.getPlatforms()) {
            platformsBuilder.append(platform.getName()).append(";");
        }
        contentValues.put(GameEntry.COLUMN_PLATFORMS, platformsBuilder.toString());

        StringBuilder genresBuilder = new StringBuilder();
        for (Genre genre : game.getGenres()) {
            genresBuilder.append(genre.getName()).append(";");
        }
        contentValues.put(GameEntry.COLUMN_GENRES, genresBuilder.toString());

        StringBuilder themesBuilder = new StringBuilder();
        for (Theme theme : game.getThemes()) {
            themesBuilder.append(theme.getName()).append(";");
        }
        contentValues.put(GameEntry.COLUMN_THEMES, themesBuilder.toString());

        StringBuilder videosBuilder = new StringBuilder();
        for (Video video : game.getVideos()) {
            videosBuilder.append(video.getName()).append("#")
                         .append(video.getVideoId()).append(";");
        }
        contentValues.put(GameEntry.COLUMN_VIDEOS, videosBuilder.toString());

        return contentValues;
    }
}

File: app/src/main/java/com/scheffer/erik/videogamewishlist/converters/GameConverter.kt
package com.scheffer.erik.videogamewishlist.converters

import com.scheffer.erik.videogamewishlist.database.databasemodels.GameDBModel
import com.scheffer.erik.videogamewishlist.database.getGenresByIdList
import com.scheffer.erik.videogamewishlist.database.getPlatformsByIdList
import com.scheffer.erik.videogamewishlist.database.getThemesByIdList
import com.scheffer.erik.videogamewishlist.models.Game
import com.scheffer.erik.videogamewishlist.models.Image
import com.scheffer.erik.videogamewishlist.models.Video

fun fromGameToGameDBModel(game: Game) =
        GameDBModel().apply {
            id = game.id
            name = game.name
            summary = game.summary
            rating = game.rating
            cover = game.cover
            platforms = game.platforms
            genres = game.genres
            themes = game.themes
            videos = game.videos
        }

fun fromGameDBModelToGame(gameDBModel: GameDBModel) =
        Game(gameDBModel.id,
                gameDBModel.name,
                gameDBModel.summary,
                gameDBModel.rating,
                gameDBModel.cover,
                gameDBModel.platforms,
                gameDBModel.genres,
                gameDBModel.themes,
                gameDBModel.videos)

fun fromProtoGameToGame(game: proto.Game) =
        Game().apply {
            id = game.id
            name = game.name
            summary = game.summary
            rating = game.rating
            cover = Image(game.cover.url, game.cover.imageId)
            videos = game.videosList.map { Video(it.name, it.videoId, game.id) }

            platforms = game.platformsList?.let { platforms ->
                getPlatformsByIdList(platforms.map(proto.Platform::getId))
            }
            genres = game.genresList?.let { genres ->
                getGenresByIdList(genres.map(proto.Genre::getId))
            }
            themes = game.themesList?.let { themes ->
                getThemesByIdList(themes.map(proto.Theme::getId))
            }
        }